src/main/java/frc/robot/commands/IntakeSequence.java

```diff
@@ -1,46 +1,54 @@
 import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.Lift.LiftPosition;
 
-public class IntakeSequence extends CommandBase {
+public class IntakeSequence extends SequentialCommandGroup {
   private Intake intake;
   private Lift lift;
 
-  public IntakeSequence(Intake intake, Lift lift) {
-    this.intake = intake;
-    this.lift = lift;
-  }
+  // deploy intake for specified amount of time
+  private ParallelRaceGroup deployIntake =
+      new RunCommand(intake::deploy, intake).withTimeout(Cal.Intake.AUTO_CLAMP_WAIT_TIME_SECONDS);
 
-  @Override
-  public void initialize() {
-    intake.deploy();
-    lift.setDesiredPosition(Lift.LiftPosition.GRAB_FROM_INTAKE);
-  }
+  // run intake infinitely (until parallel command ends)
+  private RunCommand runIntake = new RunCommand(intake::intakeGamePiece, intake);
+  
+  // trigger the lift to move to the intake position. This does not need a timeout because it is
+  // running in the parallel group, which is controlled by lift.readyToIntake()
+  private InstantCommand liftToIntakePos =
+      new InstantCommand(() -> lift.setDesiredPosition(LiftPosition.GRAB_FROM_INTAKE), lift);
 
-  @Override
-  public void execute() {
-    if (!lift.seeGamePiece()) {
-      intake.intakeGamePiece();
-    } else {
-      lift.closeGrabber();
-    }
-    // TODO fix this
-  }
+  // triggers the grabber to open when it is safe. Again, we do not care about time because it is in
+  // the parallel group.
+  private InstantCommand openGrabber = new InstantCommand(lift::openGrabber, lift);
 
-  @Override
-  public boolean isFinished() {
-    return lift.holdingGamePiece() ? true : false;
-  }
+  // run intake, move lift to intake position, and open the grabber, until the robot sees an object
+  // and the lift is in position
+  private ParallelRaceGroup intakePrep =
+      new ParallelCommandGroup(runIntake, liftToIntakePos, openGrabber).until(lift::readyToIntake);
 
-  @Override
-  public void end(boolean interrupted) {
-    if (!interrupted) {
-      lift.setDesiredPosition(Lift.LiftPosition.STARTING);
-      if (lift.clearOfIntakeZone()) { // TODO put this directly into intake
-        intake.retract();
-      }
-    }
+  // triggers the grabber to close
+  private InstantCommand closeGrabber = new InstantCommand(lift::closeGrabber, lift);
+
+  // immediately unclamps the intake.
+  private InstantCommand unclampIntake = new InstantCommand(intake::unclampIntake, intake);
+
+  // triggers the lift to move to the starting position. This does not need a timeout even though it
+  // is a longer action, because it is the final action in the sequence
+  private InstantCommand liftToStartPos =
+      new InstantCommand(() -> lift.setDesiredPosition(LiftPosition.STARTING), lift);
+
+  public IntakeSequence(Intake intake, Lift lift) {
+    this.intake = intake;
+    this.lift = lift;
+    addCommands(deployIntake, intakePrep, closeGrabber, unclampIntake, liftToStartPos);
```

Comment:
We think that it should stop intaking before closing the grabber. We also think that there should be a delay so that the grabber finishes before the intake unclamps.